clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3040,6 +3040,30 @@ static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D,
   D->addAttr(::new (S.Context) SYCLIntelNumSimdWorkItemsAttr(
         S.Context, Attr, NumSimdWorkItems));
 }
+// Handles scheduler_target_fmax_mhz
+static void handleSchedulerTargetFmaxMhzAttr(Sema &S, Decl *D,
+                                             const ParsedAttr &Attr) {
+  if (D->isInvalidDecl())
+    return;
+
+  uint32_t TargetFmaxMhz = 0;
+  const Expr *E = Attr.getArgAsExpr(0);
+  if (!checkUInt32Argument(S, Attr, E, TargetFmaxMhz, 0,
```

Comment:
Can you share the specification for this?  Are we sure there isn't a sane upper bound that we need to check?

Reply:
Thank you for the review, Erich. I am aware of the above issues. Will address them in the next patch. This is still WIP.

Reply:
@erichkeane There is no upper limit value in the spec, but 1048576 (MHz) is suggested. I was concerned about this unrealistically large value for frequency, but @jessicadavies-intel answered that unachievable values will be handled by the FPGA backend. I plan to use the suggested value as "large enough" since the type is an integer expression.